### ISSUE TITLE
Upgrade CHT-Conf version in CHT-Core

### ISSUE THREAD

**Issue by latin-panda** [medic/cht-core#7167]
**Describe the issue**
CHT-Core is using an old version of CHT-Conf, we couldn't upgrade before because of this [issue](https://github.com/medic/cht-conf/issues/396) which is already fixed.

Some app builders run the projects using the CHT-Conf defined in CHT-Core: `>> npx medic-conf [command]`. When doing this with the most recent version of CHT-Core will result of inline code errors (old eslint config) or will show errors while running some commands, etc. 

**Describe the improvement you'd like**
Upgrade CHT-Conf in CHT-Core

**Describe alternatives you've considered**
It's not a show stopper because they can install a newer version of CHT-Conf. But CHT-Core should be aligned with CHT-Conf as features become available. 

**Why Now?**
Partner(s): Internal
Reason(s): Internally needs this ticket to facilitate the release, or reduce technical debt

**Comment by aidan-plenert-macdonald** [comment]
https://github.com/medic/cht-core/pull/7340

All tests are passing now. I marked it ready for review.

**Comment by latin-panda** [comment]
To add a bit more info:
CHT Core is already using the `cht-conf`, but it also has `medic-conf` (see [package.json](https://github.com/medic/cht-core/blob/master/package.json)).

There is [a separate ticket to fix the Standard config](https://github.com/medic/cht-core/issues/7744), as part of that work the `medic-conf` will be uninstalled and we'll keep only `cht-conf`. 

However, there are some other places where we need to fix and it is this ticket work to do. Such as fixing the [grunt file](https://github.com/medic/cht-core/blob/master/Gruntfile.js#L513), fixing the[ GHA and its readme file](https://github.com/medic/cht-core/tree/master/.github/actions/deploy-with-medic-conf), test's [configs](https://github.com/medic/cht-core/blob/9ac9c1d598aa1b1949ed5444063467c4d6fec705/tests/apply-config.js#L2), and some [other appearances](https://github.com/medic/cht-core/blob/master/README.md) in different markup files.

**Comment by jkuester** [comment]
@m5r, after digging into this, I think Jenni's breakdown from the Slack thread is a good representation of how we can split up this work!  Basically we have these 4 separate, but related, pieces that we should be able to work in parallel: 

- [ ] Update `standard` tests to use cht-conf
    -  (As noted above, this has [its own issue](https://github.com/medic/cht-core/issues/7744), but I guess can ship as part of these changes.)
- [ ] Update documentation in random markdown files (and error messages from `check-app-settings-is-committed` scripts)
- [ ] Update `build-config.sh` script
- [ ] Update `apply-config.js` test

I have created the https://github.com/medic/cht-core/tree/7167_uplift_cht-conf branch from `master` and maybe we can have PRs from each sub-task into that main branch as we complete the various pieces.  

I am going to start looking at the `standard` test updates, but feel free to grab any of the others!

**Comment by m5r** [comment]
@jkuester Thanks! I'm tackling the last two items.

I've been looking for usage of the `apply-config`.js` script but I can't find anything that uses it. It doesn't seem to be working in its current state anyway

**Comment by jkuester** [comment]
FTR, I am going to hold off of https://github.com/medic/cht-core/issues/7744 since that issue may be moot if we move forwards wtih https://github.com/medic/cht-core/issues/8757
